test/integration/tcp_dump.cc

```diff
@@ -0,0 +1,75 @@
+#include "test/integration/tcp_dump.h"
+
+#include <sys/types.h>
+#include <sys/wait.h>
+#include <unistd.h>
+
+#include <fstream>
+
+#include "common/common/assert.h"
+#include "common/common/fmt.h"
+
+namespace Envoy {
+
+TcpDump::TcpDump(const std::string& path, const std::string& iface,
+                 const std::vector<uint32_t>& ports) {
+  // Remove any extant pcap file.
+  unlink(path.c_str());
+  // Derive the port filter expression.
+  std::string port_expr;
+  for (uint32_t port : ports) {
+    if (!port_expr.empty()) {
+      port_expr += " or ";
+    }
+    port_expr += "tcp port " + std::to_string(port);
+  }
+  ENVOY_LOG_MISC(debug, "tcpdumping iface {} to {} with filter \"{}\"", iface, path, port_expr);
+  // Fork a child process. We use explicit fork/wait over popen/pclose to gain
```

Comment:
I'm not that worried given this is test code, but is it safe to fork here given threads, locks, etc.? Does it happen early enough in the test process that we don't need to worry about that?

Reply:
I think it's safe given we immediately exec, but the whole pthreads+fork business is a bit of a dumpster fire. Is there a better alternative? :)